### PR TITLE
v0.1.0

### LINKED ISSUE #27: Add bounds to point mapper
Calculate the bounds and resize the map

### LINKED ISSUE #48: update read.me to explain how to use ECHO_modules to access the SBU database

### LINKED ISSUE #49: upload to pypi as package
- [x] clean up repo (e.g. #48, remove extra files)
- [ ] create pypi account and add package
- [ ] edit notebooks to use `!pip install echo_modules` instead of `!pip install git+...`

### LINKED ISSUE #51: test flexible dependencies
Currently, in the #50 branch, the dependencies are deterministic:

```
dependencies = [
	"folium==0.14.0",
	"geopandas==0.9.0",
	"ipython==7.29.0",
	"ipywidgets==7.6.5",
	"matplotlib==3.4.3",
	"numpy==1.26.3",
	"pandas==1.3.4",
	"requests==2.31.0",
	"seaborn==0.11.2"
]
```

While this has the advantage of ensuring everything will work properly, we should see if some of these can be >= instead so that it doesn't take minutes for ECHO_modules to be installed because it requires a very specific environment.

Could also look into whether these dependencies are strictly necessary (e.g. could we get away from seaborn?

### LINKED ISSUE #53: update `get_active_facilities()` to enable multi-region selections
```
elif ( region_type == 'Zip Code' ):
            sql = 'select * from "ECHO_EXPORTER" where "FAC_ZIP" = \'{}\''
            sql += ' and "FAC_ACTIVE_FLAG" = \'Y\''
            sql = sql.format( regions_selected )
            df_active = get_echo_data( sql, 'REGISTRY_ID' )
```

If `regions_selected` is a list, this won't work right. Instead, this should be updated to be like what we have for CDs:

```
elif ( region_type == 'Congressional District'):
            cd_str = ",".join( map( lambda x: str(x), regions_selected ))
            sql = 'select * from "ECHO_EXPORTER" where "FAC_STATE" = \'{}\''
            sql += ' and "FAC_DERIVED_CD113" in ({})'
            sql += ' and "FAC_ACTIVE_FLAG" = \'Y\''
            sql = sql.format( state, cd_str )
            df_active = get_echo_data( sql, 'REGISTRY_ID' )
```

### LINKED ISSUE #55: consider replacing `state_choropleth()` function with the more general `choropleth()` from the watershed notebook
See: https://github.com/edgi-govdata-archiving/ECHO_modules/blob/aa1eaf2a5ee1a1efd30eee88be343c9db7626a26/utilities.py#L310

### LINKED ISSUE #57: region filter in `get_spatial_data()` should be able to filter by multiple entries
https://github.com/edgi-govdata-archiving/ECHO_modules/blob/24c504840c1900fc6517ee7cb7224e880203e156/ECHO_modules/get_data.py#L130

Switch to something like `in "+selection+"` where selection is the region filter passed through `spatial_selector()`

### LINKED ISSUE #58: delete `selector()`?
Is it actually ever used? 

https://github.com/edgi-govdata-archiving/ECHO_modules/blob/24c504840c1900fc6517ee7cb7224e880203e156/ECHO_modules/get_data.py#L52

### LINKED ISSUE #60: add more documentation on the limitations of the ECHO database
- [ ] in the README
- [ ] in the ECHO_modules Tutorials notebook

This could include pointing to our Missing Data [notebook](https://colab.research.google.com/github/edgi-govdata-archiving/ECHO-Missing-Data/blob/main/data_disparities.ipynb) and/or Data Disparities [report](https://envirodatagov.org/wp-content/uploads/2022/09/Gaps_and_Disparities_Report.pdf)

### PR DESCRIPTION
**Pull request by ericnost**
Fixes #53, fixes #57, fixes #48, fixes #49, fixes #55, fixes #27, fixes #51, fixes #60, fixes #58 

I'm proposing to repackage ECHO_modules as something that can be delivered through pypi (pip)

I don't think this will break much^ because the repo will still be installable as a package through `pip install git+url`

^ replacing `state_choropleth_mapper()` with `choropleth()` will break the missing data notebook, but I can fix that afterwards.

Other updates:
- adds `aggregate_by_facility()` function to support `point_mapper()
- replaces `state_choropleth_mapper()` with `choropleth()`
- expands the README
- provides a [notebook](https://colab.research.google.com/drive/1EqbWz1KqBqlMc1IYMcOPbI_8Equ2Jwmr) full of tutorials/example code and queries
- basically incorporates useful tools from the work I did on the `reorganization` branch

After merging, I will set this up with pypi (#49) and also change some settings here on GitHub to automate the process of sending packages to pypi.

### CODE REVIEW

File: ECHO_modules/utilities.py [hunk 1]
```diff
@@ -376,10 +379,111 @@ def get_active_facilities( state, region_type, regions_selected ):
...
+        set(active[records.dataset.echo_type + "_IDS"]) - set(input[records.dataset.idx_field])
+        ) 
+    
+    # get rid of NaNs - probably no program IDs
+    diff = [x for x in diff if str(x) != 'nan']
+    
+    # ^ Not perfect given that some facilities have multiple NPDES_IDs
+    # Below return the full ECHO_EXPORTER details for facilities without violations, penalties, or inspections
+    diff = active.loc[active[records.dataset.echo_type + "_IDS"].isin(diff)] 
+    return diff
```
Comment (ericnost) [inline ECHO_modules/utilities.py:433]: This isn't a robust way of differentiating facilities because of the way EPA stores program IDs.

The differ function is meant to take data like:
all_facilities = [A, B, C, D, E]
facilities_with_inspections = [C, D]
and calculate:
facilities_without_inspections = [A, B, E]

But in reality, facility/program IDs are more like:
all_facilities = [A X, B Y, C, D Z, E]
facilities_with_inspections = [C, D]
So the resulting list of facilities without inspections would incorrectly be:
A X, B Y, D Z
even though D does have an inspection.

Just need a way to parse apart program IDs, ideally without having to call up the database to look at the EXP_PGM table.

**Comment by shansen5** [comment]
These all look like good and important improvements to the package.
